src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/jvm/ThreadList.java

```diff
@@ -244,8 +272,11 @@ static void runThreadDump(String pid, String[] args) {
 
     // TODO: make this map update atomic
     Util.invokePrivileged(() -> runAttachDump(pid, args));
-    runMXDump();
-
+    // oldNativeTidMap gets cleared if the attach Fails, so that the
+    // metrics collection starts as it would after a restart.
+    if (!oldNativeTidMap.isEmpty()) {
+      runMXDump();
+    }
```

Comment:
Is there a better way to indicate whether an attach failed. Map being empty can happen else where in the code maybe